### PR TITLE
feat: api_url and engine_url setting and checking

### PR DESCRIPTION
**Pull request by nokchalatte**
Clickup Task ID: CU-86enzxgd3

- Add error check and message for empty/undeclared engine URL

### CODE REVIEW

**Comment by nokchalatte** [comment]
Caveat:

- Does not cover the check for API_TOKEN, as empty API_TOKEN is allowed (and possible) during development
- Does not cover cases when API_TOKEN and/or ENGINE_URL is nonempty but wrong

For example:
image
when ENGINE_URL is http://random

-> it is difficult to test prod environment with dev env, as there are less checks in dev. I need more opinions and reports (from those who use the SDK in prod) to properly cover edge cases.

File: thanosql/_client.py [hunk 1]
```diff
@@ -23,6 +24,10 @@ def __init__(
         api_version: str = THANOSQL_API_VERSION,
         api_token: str = THANOSQL_API_TOKEN,
     ) -> None:
+        # API token can be empty but engine URL cannot
```
Comment (lazyduo) [inline thanosql/_client.py:27]: API token도 꼭 필요합니다.

https://github.com/stripe/stripe-python/blob/1657f036266412b7ef38d46c77f1b30f5455d83b/stripe/_stripe_client.py#L114

**Review by lazyduo — COMMENTED** [review]
api_token > engine_url > api_version 순서로 배치해주세요~

**Review by lazyduo — APPROVED** [review]
LGTM 👍🏻

**Review by stevensjcho — APPROVED** [review]

**Comment by lazyduo** [comment]
Task linked: [CU-86enzxgd3 API_TOKEN, ENGINE_URL input improvement](https://app.clickup.com/t/86enzxgd3)
